pallets/asset/src/lib.rs

```diff
@@ -1916,13 +1974,6 @@ impl<T: Trait> Module<T> {
             secondary_key,
         } = Identity::<T>::ensure_origin_call_permissions(origin)?;
 
-        // Check total supply here to avoid any later failure
-        Self::ensure_create_asset_parameters(&ticker, total_supply)?;
-        ensure!(
-            divisible || Self::is_unit_multiple(total_supply),
-            Error::<T>::InvalidTotalSupply
-        );
```

Comment:
Looks like you're removing a bunch of checks in general from asset creation.
Also, the checks relevant to `total_supply` need to be moved to the `_and_mint` variant.